### PR TITLE
_export: docker: supports docker option arguments

### PR DESCRIPTION
**Pull request by sonots**
Fix https://github.com/treasure-data/digdag/issues/941

Example)

```
_export:
  docker:
    image: ruby:2.5.1
    build:
      - apt install vim
    docker: /usr/local/bin/nvidia-docker
    run_options: ["-v", "/tmp:/tmp"]
    build_options: ["--pull"]
```

DESIGN: Note that DockerComandExecutor automatically adds some options of docker, thus, some options may conflict with the default options. But, this PR does not take care of it. Users have to take care of it by seeing DEBUG log.

### CODE REVIEW

**Comment by muga** [comment]
@sonots Thanks for improving for the flexibility. The feature sounds good to me. Let us take a look the details.

**Comment by muga** [comment]
@sonots sorry We don't have much time to take a look this PR but probably update back to you within next week.

File: digdag-standards/src/main/java/io/digdag/standards/command/DockerCommandExecutor.java [hunk 1]
```diff
@@ -60,23 +60,28 @@ public Process start(Path projectPath, TaskRequest request, ProcessBuilder pb)
...
+            List<String> buildOptions = optionConfig.getListOrEmpty("build", String.class);
             imageName = uniqueImageName(request, baseImageName, buildCommands);
-            buildImage(imageName, projectPath, baseImageName, buildCommands);
+            buildImage(dockerCommand, buildOptions, imageName, projectPath, baseImageName, buildCommands);
         }
         else {
             imageName = baseImageName;
             if (dockerConfig.get("pull_always", Boolean.class, false)) {
-                pullImage(imageName);
+                List<String> pullOptions = optionConfig.getListOrEmpty("pull", String.class);
```
Comment (muga) [inline digdag-standards/src/main/java/io/digdag/standards/command/DockerCommandExecutor.java]: How can "pull" option be used? If you can share your usecase for this option, please let us know.

**Comment by yoyama** [comment]
@muga I guess enable/disable "options" in system config may be one of the way.

File: digdag-standards/src/main/java/io/digdag/standards/command/DockerCommandExecutor.java (hunk 1 above)
Comment (sonots) [reply digdag-standards/src/main/java/io/digdag/standards/command/DockerCommandExecutor.java]:
```
Usage:  docker pull [OPTIONS] NAME[:TAG|@DIGEST]

Pull an image or a repository from a registry

Options:
  -a, --all-tags                Download all tagged images in the repository
      --disable-content-trust   Skip image verification (default true)
```

docker pull has only two options, -a and --disable-content-trust.
Users may use `pull: ["--disable-content-trust"]`.

**Comment by sonots** [comment]
@muga No test exist for docker options currently. I tried to add acceptance tests for docker as I pushed, but I get an error like:

```
    2019-03-10 00:49:28 +0900 [ERROR] (0037@[0:docker]+docker_build+check_touch) io.digdag.core.agent.OperatorManager: Task failed with unexpected error: java.io.IOException: Cannot run program "docker":
error=2, No such file or directory
    java.lang.RuntimeException: java.io.IOException: Cannot run program "docker": error=2, No such file or directory
        at io.digdag.standards.command.DockerCommandExecutor.buildImage(DockerCommandExecutor.java:194)
```

although the `docker_build.dig` works with:

```
$ digdag run --project digdag-tests/src/test/resources/acceptance/docker docker_build.dig
```

Could you create a first-step to add tests for docker opitons?

**Comment by sonots** [comment]
I mean that, if I use `docker: /usr/local/bin/docker` option which I added with this PR, I can make the acceptance tests work. But, how I can set `PATH` to digdag server?

**Comment by sonots** [comment]
BTW, I wondered which spec is better:

```
    options:
      run: ["-v", "/tmp:/tmp"]
      build: ["--pull"]
      pull: ["--disable-content-trust"]
```

and

```
    run_options: ["-v", "/tmp:/tmp"]
    build_options: ["--pull"]
    pull_options: ["--disable-content-trust"]
```

**Comment by sonots** [comment]
I changed the spec to the latter version because I felt it makes .dig files shorter most of cases.

File: digdag-standards/src/main/java/io/digdag/standards/command/DockerCommandExecutor.java (hunk 1 above)
Comment (sonots) [reply digdag-standards/src/main/java/io/digdag/standards/command/DockerCommandExecutor.java]: But, even I probably do not use the "pull" options. It is okay to remove it.

Comment (sonots) [reply digdag-standards/src/main/java/io/digdag/standards/command/DockerCommandExecutor.java]: I decided to remove `pull` options for now to keep it simple.

**Comment by muga** [comment]
@sonots About the option change, I also prefer your change "run_options:" and "build_options:" options instead of generic "options:". simple is good 👍  

> Could you create a first-step to add tests for _export: docker:?

Yes, let me try adding a simple test do that. Thank you for adding tests anyway.

@yoyama Ah yes, your idea sounds good to me. Let's separate it from this PR and will make improvement on another PR.

File: digdag-standards/src/main/java/io/digdag/standards/command/DockerCommandExecutor.java (hunk 1 above)
Comment (muga) [reply digdag-standards/src/main/java/io/digdag/standards/command/DockerCommandExecutor.java]: ok, I agree with your decision.

**Comment by sonots** [comment]
Tests have not passed at https://github.com/treasure-data/digdag/pull/1047, how should I proceed with this PR?

**Comment by muga** [comment]
@sonots So sorry to my slow progress of the dev on #1047.. we've recognized you're waiting for it. Let me resume it.

**Comment by muga** [comment]
@sonots Sorry to be late, I finished creating a simple test on docker-in-docker. Please take a look https://github.com/treasure-data/digdag/pull/1047

**Comment by sonots** [comment]
@muga Tests passed. PTAL.

**Comment by muga** [comment]
@sonots thanks for waiting and adding tests. the change looks good. 

@yoyama when you get a chance, please decide to include this in release v0.9.37 or 38.

**Pull request merged**